In [1]:
import pandas as pd
import numpy as np
pd.set_option("display.max_columns", None)
pd.set_option("display.max_rows", None)
HC = pd.read_excel("ASHTURR5.xlsx")
SC = pd.read_excel("ASGTURR5.xlsx")

In [2]:
HC.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6032 entries, 0 to 6031
Columns: 118 entries, IDCNTRY to SCOPE
dtypes: float64(108), int64(10)
memory usage: 5.4 MB


HC is the home context data. This is a questionnaire filled in by the parents of each student. It contains crucial information on the immigration status of students and the languages they speak. 

In [3]:
SC.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6032 entries, 0 to 6031
Columns: 155 entries, IDCNTRY to SCOPE
dtypes: float64(129), int64(26)
memory usage: 7.1 MB


SC is the Student Context data which is a questionnaire filled in by the students themselves as well as some totals on their achievement in the test. As we can see there are 6032 children who took part in this assessment in Turkey. 

Let's look at the ages these children came to the country. ASBH02B 1: Younger than 3 years old; 2: 3 to 5 years old; 3: 6 to 7 years old; 4: 8 years old or older

In [4]:
number_immigrant_children = HC[HC["ASBH02B"]< 6.0].shape[0]
print(number_immigrant_children)

227


This is about 30 more than Germany but the Turkey sample size is bigger - 6032 kids, compared to 4611 in Germany. 

Now I will merge the two datasets together for fuller comparison potential.

In [5]:
df = pd.merge(HC, SC, on='IDSTUD', how='inner')

In [11]:
df['ASBH02B'].value_counts()

6.0    5366
1.0      89
2.0      70
3.0      53
9.0      31
4.0      15
Name: ASBH02B, dtype: int64

How many children spoke Turkish before they started school? ASBH03A (1 is yes, 2 is no)

In [10]:
df['ASBH03A'].value_counts()

1.0    5080
2.0     461
9.0      83
Name: ASBH03A, dtype: int64

In [12]:
def populate_column(row):
    if row['ASBH02B'] == 1:
        return 0
    elif row['ASBH02B'] == 2:
        return 3
    elif row['ASBH02B'] == 3:
        return 6
    elif row['ASBH02B'] == 4:
        return 8
    else:
        return None  # Or any default value

# Apply the function to each row
df['MINAGEARRIVAL'] = df.apply(populate_column, axis=1)

In [15]:
df['MINAGEARRIVAL'].value_counts()

0.0    89
3.0    70
6.0    53
8.0    15
Name: MINAGEARRIVAL, dtype: int64

In [17]:
df['ASDAGE'].round().value_counts()

10.0    5092
9.0      535
11.0     369
12.0      27
14.0       4
8.0        2
13.0       2
99.0       1
Name: ASDAGE, dtype: int64

In [18]:
df['rounded_ages'] = df['ASDAGE'].round()

We want to find the arabic speaking children now

In [24]:
first_row_series2 = df.iloc[0]

# Define the file path
file_path = "first_row2.txt"

# Export the Series to a text file
first_row_series2.to_csv(file_path, header=True, index=False)

In [23]:
print(first_row_series)

IDCNTRY_x        7.920000e+02
IDPOP_x          1.000000e+00
IDGRADER_x       2.000000e+00
IDGRADE_x        4.000000e+00
WAVE_x           1.000000e+00
IDSCHOOL_x       5.001000e+03
IDCLASS_x        5.001060e+05
IDSTUD           5.001060e+07
ITLANG_HQ        3.500000e+01
LCID_HQ          1.055000e+03
ASBH01A          1.000000e+00
ASBH01B          1.000000e+00
ASBH01C          1.000000e+00
ASBH01D          1.000000e+00
ASBH01E          1.000000e+00
ASBH01F          1.000000e+00
ASBH01G          1.000000e+00
ASBH01H          1.000000e+00
ASBH01I          1.000000e+00
ASBH01J          1.000000e+00
ASBH01K          1.000000e+00
ASBH01L          1.000000e+00
ASBH01M          1.000000e+00
ASBH01N          1.000000e+00
ASBH01O          1.000000e+00
ASBH01P          1.000000e+00
ASBH01Q          1.000000e+00
ASBH01R          1.000000e+00
ASBH02A          1.000000e+00
ASBH02B          6.000000e+00
ASBH03A          1.000000e+00
ASBH03B                   NaN
ASBH03C                   NaN
ASBH03D   

In [25]:
first_row_df = df.head(1)

# Define the file path
file_path = "first_row_with_columns.txt"

# Export the DataFrame to a text file with column names
first_row_df.to_csv(file_path, header=True, index=False, sep='\t')

In [26]:
# Transpose the DataFrame to make it vertical
first_row_transposed = first_row_df.transpose()

# Define the file path
file_path = "first_row_transposed.txt"

# Export the transposed DataFrame to a text file with column names
first_row_transposed.to_csv(file_path, header=False, index=True, sep='\t')

# Optionally, print the transposed DataFrame as well
print(first_row_transposed)

                          0
IDCNTRY_x      7.920000e+02
IDPOP_x        1.000000e+00
IDGRADER_x     2.000000e+00
IDGRADE_x      4.000000e+00
WAVE_x         1.000000e+00
IDSCHOOL_x     5.001000e+03
IDCLASS_x      5.001060e+05
IDSTUD         5.001060e+07
ITLANG_HQ      3.500000e+01
LCID_HQ        1.055000e+03
ASBH01A        1.000000e+00
ASBH01B        1.000000e+00
ASBH01C        1.000000e+00
ASBH01D        1.000000e+00
ASBH01E        1.000000e+00
ASBH01F        1.000000e+00
ASBH01G        1.000000e+00
ASBH01H        1.000000e+00
ASBH01I        1.000000e+00
ASBH01J        1.000000e+00
ASBH01K        1.000000e+00
ASBH01L        1.000000e+00
ASBH01M        1.000000e+00
ASBH01N        1.000000e+00
ASBH01O        1.000000e+00
ASBH01P        1.000000e+00
ASBH01Q        1.000000e+00
ASBH01R        1.000000e+00
ASBH02A        1.000000e+00
ASBH02B        6.000000e+00
ASBH03A        1.000000e+00
ASBH03B                 NaN
ASBH03C                 NaN
ASBH03D                 NaN
ASBH03E             